llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2055,9 +2055,9 @@ MemoryDepChecker::Dependence::DepType MemoryDepChecker::isDependent(
   // stride multiplied by the backedge taken count, the accesses are independet,
   // i.e. they are far enough appart that accesses won't access the same
   // location across all loop ierations.
-  if (HasSameSize &&
-      isSafeDependenceDistance(DL, SE, *(PSE.getBackedgeTakenCount()), *Dist,
-                               MaxStride, TypeByteSize))
+  if (HasSameSize && isSafeDependenceDistance(
```

Comment:
The doc comment on isSafeDependenceDistance needs updated.  I think it's correct, but there's a difference between an exact BTC and a bound on BTC.